### PR TITLE
Exclude Maven Names Found From latest That Are Invalid

### PR DESCRIPTION
**Pull request by mikeyoung85**
This would prevent packages being added to Maven from the maven-repository.com/artifacts/latest scan if they do not have a valid name of `{group_id}:{artifact_name}`.

### CODE REVIEW

**Comment by mikeyoung85** [comment]
I don't have a way to really test this all works 100% correctly since maven-repository.com is just returning 500 errors at the moment.

File: app/models/package_manager/maven.rb [hunk 1]
```diff
@@ -210,11 +206,24 @@ def self.latest_version(versions, name)
       end
     end
 
+    def self.parse_names(page)
+      # parse the names from maven-repository.com/artifact/latest pages
```
Comment (katzj) [inline app/models/package_manager/maven.rb]: Is it worth having a test with some good and bad names to keep this from breaking?

Comment (mikeyoung85) [reply app/models/package_manager/maven.rb]: Probably. I'll need to look into how to construct the right object for the page parsing. I'll add a spec for the MavenUrl class in there too to test the `legal_name?` method I added.

File: app/models/package_manager/maven.rb [hunk 2]
```diff
@@ -210,11 +206,24 @@ def self.latest_version(versions, name)
...
+      names = page.css('tr')[1..-1]&.map do |tr|
+        tr.css('td')[0..1]&.map(&:text).join(':')
```
Comment (kbarrette) [inline app/models/package_manager/maven.rb]: `&.map` might return nil, and `.join` will bomb

File: app/models/package_manager/maven.rb [hunk 3]
```diff
@@ -210,11 +206,24 @@ def self.latest_version(versions, name)
...
+      names.select(&MavenUrl.method(:legal_name?)).uniq
+    end
+
     class MavenUrl
       def self.from_name(name)
         new(*name.split(':', 2))
       end
 
+      def self.legal_name?(name)
+        name.split(':').size == 2
```
Comment (kbarrette) [inline app/models/package_manager/maven.rb]: Looks like `name` might be nil, so this will bomb - perhaps `compact` the `names` array?

**Pull request merged**